lib/matchSourceExpression.js

```diff
@@ -0,0 +1,180 @@
+var urlModule = require('url');
```

Comment:
Agreed. I thought about doing that at some point. Most of `assets/ContentSecurityPolicy.js`, especially the parsing and normalization code, would also be a good fit for such a module.